1.Fuzzing_technology/README.md

```diff
@@ -19,73 +19,102 @@ Responsible person/main contact: Asad Hasan,
+In contrast to software testing, fuzz testing is quite the opposite.
+Fuzzing is a process of feeding malformed, mutated or unexpected inputs to a program (device or a system) and observing its behavior.
+The motivation behind this kind of testing is to discover bugs, vulnerabilities and memory leaks in a software (device or a system) for exploitation or quality improvement.
+While fuzzing is primarily used to discover and fix bugs, it can potentially be used in denial of service (DoS) attacks if certain conditions are met.
 
-The main goal of fuzzing is to make the target system behave *unexpectedly*. From the security perspective, the goal is to find and analyze those unexpected behaviors for possible exploits and figure out how to fix them. The programs that are used to perform fuzz testing are commonly called fuzzers. 
+The main goal of fuzzing is to make the target system behave *unexpectedly*. From the security perspective, the goal is to find and analyze those unexpected behaviors for possible exploits and figure out how to fix them.
+The programs that are used to perform fuzz testing are commonly called "fuzzers".
 
-Example of Resource Exhaustion with Fuzzing: Fuzzing can be used to send a large volume of specially crafted inputs to a target application, overwhelming its resources. As an example, an HTTP server could be bombarded with a flood of excessively long or malformed requests, causing it to consume excessive memory or CPU cycles, ultimately leading to a denial of service condition. 
+Example of Resource Exhaustion with Fuzzing: Fuzzing can be used to send a large volume of specially crafted inputs to a target application, overwhelming its resources. As an example, an HTTP server could be bombarded with a flood of excessively long or malformed requests, causing it to consume excessive memory or CPU cycles, ultimately leading to a denial of service condition.
 
 In this exercise you will learn:
-- Basic usage of 2 common fuzzers; Radamsa and American Fuzzy Lop (AFL). 
-- Working with AddressSanitizer, a memory error detection tool, and 
-- Valgrind, a debugging tool (can detect memory error as well). This tool is often used alongside other fuzzers.
-- Making your own fuzzer in jupyter notebook and fuzzing it
+- Basic usage of 2 common fuzzers; Radamsa and American Fuzzy Lop (AFL).
+- Working with AddressSanitizer, a memory error detection tool, and
+- Valgrind, a debugging tool (can detect memory errors as well). This tool is often used alongside other fuzzers.
+- Making your own fuzzer in Jupyter Notebook and fuzzing it
 
 ## Prerequisites
```

Comment:
Prerequisities section could be combined with preliminary tasks section to reduce duplication and make it clearer